examples/rust/opensk/src/clock.rs

```diff
@@ -12,19 +12,53 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.use opensk_lib::api::clock::Clock;
 
+use alloc::rc::Rc;
+use core::cell::Cell;
+
 use opensk_lib::api::clock::Clock;
+use wasefire::clock::Mode::Oneshot;
+use wasefire::clock::{Handler, Timer};
+
+#[derive(Default)]
+struct ClockHandler {
+    triggered: Rc<Cell<bool>>,
+}
+
+impl Handler for ClockHandler {
+    fn event(&self) {
+        self.triggered.set(true);
+    }
+}
+
+pub struct WasefireTimer {
+    timer: Timer<ClockHandler>,
+    elapsed: Rc<Cell<bool>>,
+}
+
+impl Default for WasefireTimer {
+    fn default() -> Self {
+        let elapsed = Rc::new(Cell::new(false));
+        let triggered = elapsed.clone();
+        let timer = Timer::new(ClockHandler { triggered });
```

Comment:
I'm wondering if we shouldn't try to optimize this case. Let's maybe just add a note.

```suggestion
        // This is a bit wasteful to allocate a timer that we don't need. This could be optimized later.
        let timer = Timer::new(ClockHandler { triggered });
```